languageFactories/language-factory-core/src/main/scala/cromwell/languages/util/ImportResolver.scala

```diff
@@ -12,86 +13,136 @@ import com.softwaremill.sttp.asynchttpclient.cats.AsyncHttpClientCatsBackend
-    directoryValidation flatMap { directoryResolver(_).run(path) }
+    override def name: String = s"relative to directory $absolutePathToDirectory (without escaping $dontEscapeFrom)"
   }
 
-  lazy val localFileResolver: ImportResolver = CheckedAtoB.fromErrorOr { path =>
-    Try(Paths.get(Paths.get(".").resolve(path).toFile.getCanonicalPath)).toErrorOr flatMap { absolutePathToFile =>
-      val file = File(absolutePathToFile)
-      if (file.exists) {
-        File(absolutePathToFile).contentAsString.validNel
-      } else {
-        s"Import file not found: $path".invalidNel
-      }
+  def zippedImportResolver(zippedImports: Array[Byte]): ErrorOr[ImportResolver] = {
+    LanguageFactoryUtil.validateImportsDirectory(zippedImports) map { dir =>
+      DirectoryResolver(dir, Option(dir.toJava.getCanonicalPath))
     }
   }
 
-  def specificFileResolver(filePath: Path): ImportResolver = CheckedAtoB.fromErrorOr { path =>
-    if (path == filePath.toAbsolutePath.toString || path == filePath.name) {
-      if (filePath.exists) {
-        File(filePath.toAbsolutePath.pathAsString).contentAsString.validNel
-      } else {
-        s"Import file not found: $path".invalidNel
-      }
-    } else {
-      s"'$path' did not match specific file ${filePath.toAbsolutePath.pathAsString}".invalidNel
+  case class HttpResolver(relativeTo: Option[String] = None, headers: Map[String, String] = Map.empty) extends ImportResolver {
+    override def name: String = {
+      s"http importer${relativeTo map { r => s" (relative to $r)" } getOrElse ""}"
     }
-  }
 
-  /**
-    * Given a string representing an http(s) url, retrieve the contents
-    */
-  def httpResolver: ImportResolver =
-    httpResolverWithHeaders(Map.empty[String,String])
+    override def innerResolver(str: String, currentResolvers: List[ImportResolver]): Checked[ResolvedImportBundle] = {
 
-  /**
-    * Given a string representing an http(s) url, retrieve the contents
-    * using the supplied headers (which can be an empty map)
-    */
-  def httpResolverWithHeaders(headers: Map[String,String]): ImportResolver = CheckedAtoB.fromCheck { str: String =>
+      def newResolverList(newRoot: String): List[ImportResolver] = {
+        val rootWithoutFilename = newRoot.split('/').init.mkString("", "/", "/")
+        List(
+          HttpResolver(relativeTo = Some(canonicalize(rootWithoutFilename)), headers)
+        )
+      }
 
-    implicit val sttpBackend = AsyncHttpClientCatsBackend[IO]()
+      def canonicalize(str: String): String = {
+        val url = new URL(str)
+        val uri = new URI(url.getProtocol, url.getUserInfo, url.getHost, url.getPort, url.getPath, url.getQuery, url.getRef)
+        uri.toString
+      }
 
-    val responseIO : IO[Response[String]] =
-      sttp.get(uri"$str").headers(headers).send()
+      def root(str: String): String = {
+        val url = new URL(str)
+        val uri = new URI(url.getProtocol, url.getUserInfo, url.getHost, url.getPort, "", "", "")
```

Comment:
Perhaps add a test to cover this line and make sure this method is doing exactly what you want.

I'm pretty sure those empty strings `""` should each be `null`. As is, the `toString` below returns strings ending with `?#`.

If this is expected behavior then please leave a comment explaining why.